### PR TITLE
switch to new registrator and drop CI support for julia 0.7

### PR DESCRIPTION
**Pull request by johnnychen94**
more details see https://github.com/JuliaImages/Images.jl/issues/798

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Requires = "ae029012-a4dd-5104-9daa-d747884805df"
+
+[compat]
+ColorTypes = ">= 0.7.4"
+julia = ">= 1.0"
+
+[extras]
+ImageMagick = "6218d12a-5da1-5696-b52f-db25d2ecc6d1"
+PaddedViews = "5432bcbf-9aad-5242-b902-cca2824c8663"
+QuartzImageIO = "dca85d43-d64c-5e67-8c65-017450d5d020"
```
Comment (Evizero) [inline Project.toml]: I am guessing the `@osx` marcos etc are gone? should we maybe just stcik with ImageMagick in this case?

Comment (johnnychen94) [reply Project.toml]: According to https://github.com/JuliaLang/Pkg.jl/issues/165 this seems not implemented yet.

File: Project.toml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+Colors = "5ae59095-9a9b-59fe-a467-6f913c188581"
+FileIO = "5789e2e9-d7fb-5bc7-8068-2c6fae9b9549"
+FixedPointNumbers = "53c48c17-4a7d-5ca2-90c5-79b7896eea93"
+ImageCore = "a09fc81d-aa75-5fe9-8630-4744c3626534"
+OffsetArrays = "6fe1bfb0-de20-5000-8ca7-80f57d26f881"
+Requires = "ae029012-a4dd-5104-9daa-d747884805df"
+
+[compat]
+ColorTypes = ">= 0.7.4"
+julia = ">= 0.7"
```
Comment (timholy) [inline Project.toml]:
```suggestion
julia = ">= 1"
```

**Comment by johnnychen94** [comment]
Is this problem caused by https://github.com/JuliaIO/ImageMagick.jl/blob/40d725a1f0bf9051585df2b0a9bf62a579c9955f/REQUIRE#L6?

Perhaps we need to bring back 0.7 compatibility

**Comment by johnnychen94** [comment]
Looks like either ImageMagick or ImageShow needs a version bump first to solve the version conflict.

**Comment by johnnychen94** [comment]
### Option1:

Adding a compatibility restriction to `ImageCore = "< 0.8"` solves the version conflict

1. bump to `v0.1.3` as a temporary/transitory version, I think that can also make https://github.com/JuliaIO/ImageMagick.jl/pull/152 work, perhaps there should be a similar operation on ImageMagick side cc: @ianshmean

2. When the conflict resolved, drop the `ImageCore` restriction and bump to `0.2.0`

Still keep compatibility to julia 0.7 until `ImageShow v0.2.0`

### Option2:
drop the `ImageCore` restrction, merge and bump the version to `0.2.0` regardless of the CI failure.

**Comment by johnnychen94** [comment]
ping @timholy

**Comment by ianshmean** [comment]
https://github.com/JuliaIO/ImageMagick.jl/pull/152 is now merged and released. Awaiting merging on General

**Comment by johnnychen94** [comment]
Just noticed it, thanks!
In that case ImageShow can directly upgrade to `0.2.0

**Comment by RalphAS** [comment]
Fixed by above-mentioned PR to General.

**Pull request merged**